Unprovoked aggression means open season

AsFoxy opened the pull request:
Attacking a non bandit non-currently-open-season player causes open season on said attacker, meaning no humanity is lost upon shooting them.

Open season expires after X amount of time which can be set by the server admin. Default is 10 minutes. The timer is reset for each act of unprovoked aggression.

Due to technical limitations only actual impacts are counted. I.e. if a player attacks unprovoked and misses, this does not cause open season, and in fact if the player attacked then retaliates and hits, they themselves might be subject to open season.

Big thanks to @HeroSmurf

AsFoxy committed:
OpenTarget

AsFoxy committed:
OpenTarget

AsFoxy committed:
OpenTarget

AsFoxy committed:
OpenTarget

AsFoxy committed:
OpenTarget

AsFoxy committed:
OpenTarget

KPanicEKS commented:
I love this idea. Unfortunately we gotta think about what happens if they
relog while they are in open season. Is there a flag that can be set in the
DB that gets flagged and would just reset the timer if they tried to log
back in?

On Wed, May 27, 2015 at 6:05 PM, Foxy notifications@github.com wrote:

> Attacking a non bandit non-currently-open-season player causes open season
> on said attacker, meaning no humanity is lost upon shooting them.
> 
> Open season expires after X amount of time which can be set by the server
> admin. Default is 10 minutes. The timer is reset for each act of unprovoked
> aggression.
> 
> ## Big thanks to @HeroSmurf https://github.com/HeroSmurf
> 
> You can view, comment on, or merge this pull request online at:
> 
>   https://github.com/DayZMod/DayZ/pull/480
> Commit Summary
> - OpenTarget
> - OpenTarget
> - OpenTarget
> - OpenTarget
> - OpenTarget
> - OpenTarget
> 
> File Changes
> - _M_ SQF/Missions/dayz_1337.chernarus/init.sqf
>   https://github.com/DayZMod/DayZ/pull/480/files#diff-0 (2)
> - _M_ SQF/dayz_code/compile/fn_damageHandler.sqf
>   https://github.com/DayZMod/DayZ/pull/480/files#diff-1 (29)
> - _M_ SQF/dayz_code/compile/player_death.sqf
>   https://github.com/DayZMod/DayZ/pull/480/files#diff-2 (21)
> - _M_ SQF/dayz_code/init/publicEH.sqf
>   https://github.com/DayZMod/DayZ/pull/480/files#diff-3 (3)
> - _M_ SQF/dayz_code/init/variables.sqf
>   https://github.com/DayZMod/DayZ/pull/480/files#diff-4 (3)
> - _M_ SQF/dayz_code/system/player_spawn_2.sqf
>   https://github.com/DayZMod/DayZ/pull/480/files#diff-5 (8)
> - _M_ SQF/dayz_server/compile/server_sendToClient.sqf
>   https://github.com/DayZMod/DayZ/pull/480/files#diff-6 (9)
> 
> Patch Links:
> - https://github.com/DayZMod/DayZ/pull/480.patch
> - https://github.com/DayZMod/DayZ/pull/480.diff
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/DayZMod/DayZ/pull/480.

R4Z0R49 commented:
that should be the easy bit. in playersync (https://github.com/DayZMod/DayZ/blob/Development/SQF/dayz_server/compile/server_playerSync.sqf#L166) add a getvea for the opentarget

R4Z0R49 commented:
then to reload after login 
https://github.com/DayZMod/DayZ/blob/Development/SQF/dayz_server/compile/server_playerSetup.sqf#L70
https://github.com/DayZMod/DayZ/blob/Development/SQF/dayz_server/compile/server_playerSetup.sqf#L254

then in 
https://github.com/DayZMod/DayZ/blob/Development/SQF/dayz_code/system/player_monitor.fsm#L669

Add it all back to the client

AsFoxy committed:
Fixed addon load order

AsFoxy committed:
dayz_equip to dayz_code config move

AsFoxy committed:
Cleanup

AsFoxy committed:
Disabled Dayz STANAG model

Disabled Dayz model in favor of community configuration project model.
No change from 1.8.6.

AsFoxy committed:
dayz_equip and dayz_weapons to dayz_code config move
